Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -801,6 +801,27 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
     }
 
     internal func populateTargetTrie(callback: @escaping (Result<Void, Error>) -> Void = { _ in }) {
+        // Check to see if there is any data before submitting task to queue because otherwise it's no-op anyway
+        do {
+            let numberOfCollections: Int = try self.executeStatement("SELECT COUNT(*) FROM \(Self.packageCollectionsTableName);") { statement in
+                let row = try statement.step()
+                guard let count = row?.int(at: 0) else {
+                    throw StringError("Failed to get count of \(Self.packageCollectionsTableName) table")
+                }
+                return count
+            }
+            // No collections means no data, so no need to populate target trie
+            guard numberOfCollections > 0 else {
```

Comment:
is there an edge case here for the last collection being deleted where we would want to reset the trie?

Reply:
Adding/deleting a collection updates the trie. The method here is only called by the initializer to seed the trie before any other db updates.